### ISSUE TITLE
statement with no effect warning

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust#42280]
This is an enhancement request. This little program currently compiles with no warnings:

```
fn main() {
    "sss";
    10;
    10 == 10;
}
```

Clippy gives three "statement with no effect" warnings. I think such warning should be raised by the rustc compiler too.

**Comment by Mark-Simulacrum** [comment]
Thanks! This is actually a duplicate of https://github.com/rust-lang/rust/issues/32108. Either way, I think this would need an RFC to be implemented due to the nature of lints, but I'm uncertain.

**Comment by steveklabnik** [comment]
> Either way, I think this would need an RFC to be implemented due to the nature of lints, but I'm uncertain.

Yes, new lints require RFCs.
